txpool/pool.go

```diff
@@ -1184,6 +1190,15 @@ func (p *TxPool) setBaseFee(baseFee uint64) (uint64, bool) {
 	return p.pendingBaseFee.Load(), changed
 }
 
+func (p *TxPool) setBlobFee(blobFee uint64) (uint64, bool) {
+	changed := false
+	if blobFee > 0 {
+		changed = blobFee != p.pendingBlobFee.Load()
```

Comment:
“Read then write” on atomic - it’s race-condition. Please use  CompareAndSwap operation.